### ISSUE TITLE
SyncedVideoPlayer Module : Execution Plan

### ISSUE THREAD

**Issue by udhaymanhas** [pesto-students/juntos-ninja-2-beta#22]
SyncedVideoPlayer(user, videoId, channel, roomId)
- start()
- stop()
- play()
- pause()

Once the youtube player is ready, an instance of SyncedVideoPlayer will be created. The syncing of the video player will start in the party room, the user, videoId, channel, and roomId will already be available. Once the instance of SyncedVideoPlayer is available, any change on the video player by the host of the party will fire the respective functions available under SyncedVideoPlayer.

Each function (listed above) will be triggered by a change in the video player status and the function will send the sync information to the backend service via WebSockets.

 For the video player, I will implement a simple page Youtube iFrame.
 For WebSockets, I will be creating a simple nodejs project which enables the required WebSockets initialization on the client-side and further communication.
 
 There might be a few other changes regarding the above implementation.
 
 Also, I think that the front end has to be served via the same backend code as per the requirements of the socket communication. (The client-side app and client-side socketio.js are served from the backend implementing the socketio.)
Will need to check if there is another way that enables us to do it separately.

**Comment by chirgjn** [comment]
This seems fine.
I'd recommend that you implement only the `start` functionality first and raise a PR with that.

**Comment by udhaymanhas** [comment]
Execution Plan 2: SyncedVideoPlayer

- Implement a sample Youtube Player page with a fixed video file for now.
- Create a start button on the same page that starts the playback on all connected users as follows
   - Create socket connection
   - Share video and playback details
   - Start syncing process by sharing progress

Expected timeline: 11th June EOD

There may be a change in the number and name of functions, I will add the description alongside the functions in the comment and we can discuss improving the changes once the basic functional goal is achieved.

**Comment by chirgjn** [comment]
Can you elaborate on the following:

>    - Start syncing process by sharing progress

**Comment by udhaymanhas** [comment]
- Share video and playback details
   -  The host will select the video(a default video for now). 
   -  The host will then start the party
   -  The start party will signal other connected users by sharing the selected videoId and startSeconds
 

- Start syncing process by sharing progress
   - Sharing the running playback seconds on both host and viewer side with dummy backend
   - The dummy backend will check for any discrepancy(a simple +- 150ms check for now)

**Comment by chirgjn** [comment]
>    -  The host will select the video(a default video for now). 
>    -  The host will then start the party

This is not the responsibility of this module. Check https://en.wikipedia.org/wiki/Single-responsibility_principle

> - The start party will signal other connected users by sharing the selected videoId and startSeconds

How?

> - Sharing the running playback seconds on both host and viewer side with dummy backend

How?

**Comment by udhaymanhas** [comment]
> > * The host will select the video(a default video for now).
> > * The host will then start the party
> 
> This is not the responsibility of this module. Check https://en.wikipedia.org/wiki/Single-responsibility_principle

 I do not understand what the `start()` function is supposed to do? What is it supposed to start?

> > * The start party will signal other connected users by sharing the selected videoId and startSeconds
> 
> How?

By sharing the details(videoId and startSeconds) over a socket event.


> > * Sharing the running playback seconds on both host and viewer side with dummy backend
> 
> How?

(We discussed this when I informed you about `iFrame` vs `HTML5 <video>` capability to share back the current video progress from the player) A function running continuously and sending the info(current playback second) via socket.

**Comment by chirgjn** [comment]
Postponed. We need more detailed requirements to nail down the plan.
